service-tests/chocolatey.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+const Joi = require('joi');
+const ServiceTester = require('./runner/service-tester');
+const {
+  isMetric,
+  isVPlusDottedVersionNClauses
+} = require('./helpers/validators');
+
+const t = new ServiceTester({ id: 'chocolatey', title: 'Chocolatey' });
+module.exports = t;
+
+
+// downloads
+
+t.create('total downloads (valid)')
+  .get('/dt/scriptcs.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'chocolatey',
+    value: isMetric,
+  }));
+
+t.create('total downloads (not found)')
+  .get('/dt/not-a-real-package.json')
+  .expectJSON({name: 'chocolatey', value: 'not found'});
+
+t.create('total downloads (connection error)')
+  .get('/dt/scriptcs.json')
+  .networkOff()
+  .expectJSON({name: 'chocolatey', value: 'inaccessible'});
+
+t.create('total downloads (unexpected response)')
+  .get('/dt/scriptcs.json')
+  .intercept(nock => nock('https://www.chocolatey.org')
+    .get("/api/v2/Packages()?$filter=Id%20eq%20%27scriptcs%27%20and%20IsLatestVersion%20eq%20true")
+    .reply(200, "{{{{{invalid json}}")
+  )
+  .expectJSON({name: 'chocolatey', value: 'invalid'});
+
+
+// version
```

Comment:
"version" and "pre version" badge can have different colors (yellow, orange or blue). Would you like to test these cases? Color value can be check using this approach https://github.com/badges/shields/blob/8ad176d014bdddc95f06a95ae4ff82c7697bb569/service-tests/github.js#L39-L40

Reply:
I had not realised that it we had an endpoint to test the colours but I now see you added this in #1190 - this would probably be helpful in some other tests.

I think it would also be helpful to add an example to the [testing docs](https://github.com/badges/shields/blob/master/service-tests/README.md) to explain this feature for other contributors.

Reply:
I agree with you. I will try to add such example.